mautic/mautic
# Fix limit/batch for broadcast email command

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | [mautic/mautic-documentation#... ](https://github.com/mautic/mautic-documentation/pull/313)
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



#### Steps to test this PR:
Broadcast command for emails process incorrect limit/batch parameters during the sending. This PR fixed it and make same way of processing these parameters like SMS channel.

For example

`mautic:broadcast:send --channel=email --limit=10 --batch=2`  - before process all emails in batch 10. We expect process 10 emails with command in batch 2



1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create segments email and try many combinations

`mautic:broadcast:send --channel=email --limit=10 --batch=2` - should process 10emails in batch 2 (**before process all emails in batch 10**)
`mautic:broadcast:send --channel=email --batch=2` - should process all emails in batch 2
`mautic:broadcast:send --channel=email --limit=10` - should process 10 emails per  command without batch processing
`mautic:broadcast:send --channel=email` - should process all email without batch processing




# Backwards compatibility breaks
If you use limit and batch parameters for `mautic:broadcast:send`, you should  re-arrange it. Mention it in UPGRADE-5.0.md
 https://github.com/mautic/mautic/pull/11127/files#diff-f125ea0a4ebc7600394bee94245a513bc6d5b2908c5b8fe3644bcef0fe1bf9b3R6




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11127"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
